pkg/assembler/grammar.go

```diff
@@ -68,7 +68,7 @@ type DoubleDeref struct {
 
 type Offset struct {
 	Sign  string `@("+" | "-")`
-	Value *int   `@Int`
+	Value *int   `@('-'? Int)`
```

Comment:
Nice fix! I wasn't aware you could do that.
I think you can use double quotes in `"-"` to use the same format as the  rest of the code, but other than that, it looks good

Reply:
Good call.
Fixed.